airbyte-webapp/src/packages/cloud/services/dbtCloud.ts

```diff
@@ -0,0 +1,116 @@
+// This module is for the business logic of working with dbt Cloud webhooks.
+// Static config data, urls, functions which wrangle the APIs to manipulate
+// records in ways suited to the UI user workflows--all the implementation
+// details of working with dbtCloud jobs as webhook operations, all goes here.
+// The presentation logic and orchestration in the UI all goes elsewhere.
+//
+// About that business logic:
+// - for now, the code treats "webhook operations" and "dbt Cloud job" as synonymous.
+// - custom domains aren't yet supported
+
+import isEmpty from "lodash/isEmpty";
+
+import { OperatorType, WebBackendConnectionRead, OperationRead } from "core/request/AirbyteClient";
+import { useWebConnectionService } from "hooks/services/useConnectionHook";
+import { useCurrentWorkspace } from "hooks/services/useWorkspace";
+import { useUpdateWorkspace } from "services/workspaces/WorkspacesService";
+
+export interface DbtCloudJob {
+  account: string;
+  job: string;
+  operationId?: string;
+}
+const dbtCloudDomain = "https://cloud.getdbt.com";
+const webhookConfigName = "dbt cloud";
+const executionBody = `{"cause": "airbyte"}`;
+const jobName = (t: DbtCloudJob) => `${t.account}/${t.job}`;
+
+const toDbtCloudJob = (operation: OperationRead): DbtCloudJob => {
+  const { operationId } = operation;
+  const { executionUrl } = operation.operatorConfiguration.webhook || {};
+
+  const matches = (executionUrl || "").match(/\/accounts\/([^/]+)\/jobs\/([^]+)\//);
+
+  if (!matches) {
+    throw new Error(`Cannot extract dbt cloud job params from executionUrl ${executionUrl}`);
+  } else {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_fullUrl, account, job] = matches;
+
+    return {
+      account,
+      job,
+      operationId,
+    };
+  }
+};
+const isDbtCloudJob = (operation: OperationRead): boolean =>
+  operation.operatorConfiguration.operatorType === OperatorType.webhook;
+
+export const useSubmitDbtCloudIntegrationConfig = () => {
+  const { workspaceId } = useCurrentWorkspace();
+  const { mutate: updateWorkspace } = useUpdateWorkspace();
+
+  return (authToken: string, singleTenantUrl?: string) => {
+    const domain = singleTenantUrl || dbtCloudDomain;
+    const validationUrl = `${domain}/api/v2/accounts`;
+
+    updateWorkspace({
+      workspaceId,
+      webhookConfigs: [
+        {
+          name: webhookConfigName,
+          authToken,
+          validationUrl,
+        },
+      ],
+    });
+  };
+};
+
+export const useDbtIntegration = (connection: WebBackendConnectionRead) => {
+  const workspace = useCurrentWorkspace();
+  const { workspaceId } = workspace;
+  const connectionService = useWebConnectionService();
+
+  // TODO extract shared isDbtWebhookConfig predicate
+  const hasDbtIntegration = !isEmpty(workspace.webhookConfigs?.filter((config) => /dbt/.test(config.name || "")));
```

Comment:
I got paranoid about locking in our ability to edit the webhook name if we decide to tweak the naming convention, but I think you're right that it would be cleaner just checking for the actual value, since it's not user-supplied

Reply:
I'm not sure whether we meant the same thing, so just for clarification, I suggested just replacing the regexp by an includes not the whole filter statement, i.e.:

```suggestion
  const hasDbtIntegration = !isEmpty(workspace.webhookConfigs?.filter((config) => config.name?.includes("dbt")));
```